parquet-hadoop/src/main/java/org/apache/parquet/hadoop/DictionaryPageReader.java

```diff
@@ -78,47 +91,32 @@ public DictionaryPage readDictionaryPage(ColumnDescriptor descriptor) {
       return rowGroup.readDictionaryPage(descriptor);
     }
 
-    String dotPath = Strings.join(descriptor.getPath(), ".");
+    String dotPath = String.join(".", descriptor.getPath());
     ColumnChunkMetaData column = columns.get(dotPath);
     if (column == null) {
       throw new ParquetDecodingException(
-          "Cannot load dictionary, unknown column: " + dotPath);
+          "Failed to load dictionary, unknown column: " + dotPath);
     }
 
-    if (cache.containsKey(dotPath)) {
-      return cache.get(dotPath);
-    }
+    return cache.computeIfAbsent(dotPath, key -> {
```

Comment:
Is this method threadsafe? Since we're removing the locking mechanism.

Reply:
Yes.  Since this is `ConcurrentHashMap` this is thread-safe. 

https://docs.oracle.com/javase/8/docs/api/java/util/concurrent/ConcurrentHashMap.html#computeIfAbsent-K-java.util.function.Function-

It also has a more fine-grained locking mechanism.  So, depending on the use case, it may be beneficial in regards to performance. i.e, more than one thread can access the structure as long as they're not both accessing the same area of the Map.

Reply:
Missed the `Concurrent` part :-) Thanks for picking this up